src/odemis/driver/tescan.py

```diff
@@ -1127,12 +1170,22 @@ def _doMoveAbs(self, pos):
             logging.debug("Stage move to %s completed", pos)
             self._updatePosition()
 
-    def _doMoveRel(self, shift):
-        # Compute the position in absolute coordinates
+    def _doMoveRel(self, shift: dict):
+        """
+        Request the position in absolute coordinates and create new positions after applying shift values
+        :param shift: shift of linear axes in mm or rotational axes in radians
+        """
         self._updatePosition()
         pos = self._position.copy()
+
+        # add the requested change to the current position
         for axis, change in shift.items():
-            pos[axis] += change
+            if axis not in {"rz"}:
+                pos[axis] += change
+            # different approach for the Rz axis as it can fully turn back and forth
```

Comment:
As you use the stage limits given by the SEM software, you cannot have so much expectations here.
I'd suggest to use the same code as we've used in other drivers (eg, xt_client):
```
if axis in ("rx", "rz"):
    rng = self.axes[axis].range
    if  util.almost_equal(rng[1] - rng[0], 2 * math.pi):
                pos[axis] = (pos[an] - rng[0]) % (2 * math.pi) + rng[0]
```